pycqed/analysis/tools/data_manipulation.py

```diff
@@ -286,6 +300,60 @@ def count_error_fractions(trace):
     return no_err_counter, single_err_counter, double_err_counter, zero_counter, one_counter
 
 
+def mark_errors_flipping(events):
+    '''
+    Marks error fractions
+    '''
+    single_errors = np.zeros(len(events)-1)
+    double_errors = np.zeros(len(events)-2)
+
+    for i in range(len(events)-1):
+            # A single error is associated with a qubit error
+            if (events[i] == events[i+1]):
+                single_errors[i] = 1
+                if i < (len(events)-2):
+                    # two identical outcomes equal to one
+                    if (events[i] == events[i+2]):
+                        double_errors[i] = 1
+    return single_errors, double_errors
+
```

Comment:
again, this can be expressed shorter and faster with numpy expressions:

```
single_errors = (events[1:] + events[:-1])%2
double_errors = (events[2:] + events[:-2])%2
```

Reply:
Tested, not identical. 
There probably is an elegant and fast way to do this but I will not do that now.